ERCS/erc-7846.md

```diff
@@ -0,0 +1,168 @@
+--- 
+eip: 7846
+title: Wallet Connection API  
+description: Adds JSON-RPC method for requesting wallet connection with modular capabilities.  
+author: Conner Swenberg (@ilikesymmetry), Jake Moxey (@jxom).
+discussions-to: https://ethereum-magicians.org/t/erc-7846-wallet-connection-api/22245
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-12-15
+---
+
+## Abstract
+
+This ERC introduces a new wallet connection RPC method focused on extensibility. It leverages the modular capabilities approach defined in [ERC-5792](https://eips.ethereum.org/EIPS/eip-5792#wallet_getcapabilities) to streamline connections and authentication into a single interaction.
+
+## Motivation
+
+Current standards like `eth_requestAccounts` and `personal_sign` lack extensibility and require separate interactions for connection and authentication. This results in added complexity for both users and developers. A unified and extensible RPC can enhance user experience, simplify development, and prepare for increasing sophistication in wallet interactions.
+
+## Specification
+
+### `wallet_connect`
+
+Request the user to connect a single account and optionally confirm chain support and add capabilities.
+
+#### RPC Specification
+
+For each chain requested by the app, wallets MUST return a mapped capabilities object if the chain is supported and SHOULD return an empty object if no capabilities exist. Wallets MUST NOT return mapped capabilities objects for chains they do not support. If an app does not declare chains they would like to confirm support for, the wallet can return any chains it wishes to declare support for.
+
+```typescript
+type WalletConnectParams = [{
```

Comment:
```suggestion
type WalletConnectRequest = [{
```

Nit: But more in-line with how 1993 handles things. I know 5792 didn't take that approach.